packages/owlbot-bootstrapper/common-container/mono-repo.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Language} from './interfaces';
+import {Octokit} from '@octokit/rest';
+import {
+  getBranchName,
+  openABranch,
+  openAPR,
+  cmd,
+  checkIfGitIsInstalled,
+} from './utils';
+
+/**
+ * Monorepo class
+ *
+ * @param language the Language of the particular monorepo
+ * @param repoToCloneUrl the url for the repo to clone without the preceeding https://, e.g., just github.com/googleapis/nodejs-kms.git
+ * @param repoName the name for the repo to clone, i.e., nodejs-kms
+ * @param githubToken a short-lived access Github access token
+ * @param octokit an instance of Octokit
+ */
+export class MonoRepo {
+  language: Language;
+  repoToCloneUrl: string;
+  repoName: string;
+  githubToken: string;
+  octokit: Octokit;
+
+  constructor(
+    language: Language,
+    repoToCloneUrl: string,
+    githubToken: string,
+    octokit: Octokit
+  ) {
+    this.language = language;
+    this.repoToCloneUrl = repoToCloneUrl;
+    // Get the repo name from the repoToCloneUrl, i.e. github.com/googleapis/nodejs-kms.git becomes nodejs-kms
+    this.repoName = repoToCloneUrl.split('/')[2].split('.')[0];
+    this.githubToken = githubToken;
+    this.octokit = octokit;
+  }
+
+  /**
+   * Clones a repo from github
+   *
+   * @param githubToken a short-lived access Github access token
+   * @param repoToCloneUrl from where to clone the repo
+   * @param directoryPath where to clone the repo to
+   */
+  public async _cloneRepo(
```

Comment:
nit: does this need to be public (the underscore-led name implies it doesn't)

Reply:
It doesn't, but it's making testing easier. Do you have any recs?

Reply:
This code is now an implementation detail of the common container so it's less of a big deal.

Visibility in typescript is a bit hard because anything exported from any module is technically public, but generally, we can assume that a developer should only `import {something} from 'my-package-name'`. Public methods on an exported class can be unintentionally exposed if the exported class is considered part of the public interface (again, not a big concern for this package). Since this method also requires no instance methods, you could move it to be an exported (from this module, but not the main package) function within this file so it can be tested.

Reply:
Another option is to test the public methods that use this code. Generally, testing simpler private methods is not advised as it's an implementation detail of the real interface and thus you're coupling your tests to the implementation.